src/command_modules/azure-cli-vm/azure/cli/command_modules/vm/_validators.py

```diff
@@ -3,8 +3,110 @@
+        from azure.mgmt.resource.resources import ResourceManagementClient
+        from azure.cli.core.commands.client_factory import get_mgmt_service_client
+
+        resource_client = get_mgmt_service_client(ResourceManagementClient)
+        vnet_client = get_mgmt_service_client(NetworkManagementClient).virtual_networks
+
+        rg = resource_client.resource_groups.get(ns.resource_group_name)
+        location = ns.location or rg.location # pylint: disable=no-member
+
+        # find VNET in target resource group that matches the VM's location
+        vnet = next((v for v in vnet_client.list(rg.name) if v.location == location), None) # pylint: disable=no-member
+
+        # TODO: if none found, find qualifying VNET in subscription (like portal)
+        # Current ARM templates do not support this
+        #if not vnet:
+        #    # if not matching VNET in the resource group, try the subscription
+        #    vnet = next((v for v in vnet_client.list_all() if v.location == location), None)
+
+        if vnet:
+            try:
+                ns.subnet_name = vnet.subnets[0].name
+                ns.virtual_network = vnet.name
+                ns.virtual_network_type = 'existingName'
+            except KeyError:
+                pass
+
+def validate_default_storage_account(namespace):
+    ns = namespace
+    if not ns.storage_account and not ns.storage_account_type:
+        from azure.mgmt.storage import StorageManagementClient
+        from azure.mgmt.resource.resources import ResourceManagementClient
+        from azure.cli.core.commands.client_factory import get_mgmt_service_client
+
+        resource_client = get_mgmt_service_client(ResourceManagementClient)
+        storage_client = get_mgmt_service_client(StorageManagementClient).storage_accounts
+
+        rg = resource_client.resource_groups.get(ns.resource_group_name)
+        location = ns.location or rg.location # pylint: disable=no-member
+
+        sku_tier = 'Premium' if 'Premium' in ns.storage_type else 'Standard'
+        # find storage account in target resource group that matches the VM's location
+        account = next((a for a in storage_client.list_by_resource_group(ns.resource_group_name)
+                        if a.sku.tier.value == sku_tier and a.location == location), None)
+
+        # TODO: if none found, find qualifying storage account in subscription (like portal)
+        # Current ARM templates do not support this
+        #if not account:
+        #    # if not matching storage account in the resource group, try the subscription
+        #    account = next((a for a in storage_client.list()
+        #                    if a.sku.tier.value == sku_tier and a.location == location), None)
+
+        if account:
+            ns.storage_account = account.name
+            ns.storage_account_type = 'existingName'
+        else:
+            ns.storage_account = 'vhd{}{}'.format(str(int(math.ceil(time.time())))[:9],
```

Comment:
Oh, you are talking about the randomized GUID for OS disk and storage account? I thought you meant the defaulting logic between storage account and vnet. :grin: This is one of the limitations of Github commenting.

Anyways, yes I agree--I'll refactor that into a method.